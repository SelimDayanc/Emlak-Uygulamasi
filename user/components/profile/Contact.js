/**
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {ScrollView, StyleSheet, Text, View} from 'react-native';
import colors from '../../../common/colors';
import Ionicons from 'react-native-vector-icons/Ionicons';
import Separator from './../../../components/Separator';

export default class Contact extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
  };

  render() {
    const {user} = this.props;
    return (
      <ScrollView style={styles.container}>

        {user.mobile &&
          <View>
            <View style={styles.rowContainer}>

              <Text style={styles.label}>MOBILE</Text>
              <View style={styles.content}>
                <Ionicons
                  name="md-phone-portrait"
                  size={20}
                  style={styles.icon}
                />
                <Text style={styles.name}>{user.mobile}</Text>
              </View>
            </View>
            <Separator style={{marginVertical: 20}} />
          </View>}

        {user.email &&
          <View>

            <View style={styles.rowContainer}>
              <Text style={styles.label}>EMAIL</Text>
              <View style={styles.content}>
                <Ionicons name="ios-mail" size={22} style={styles.icon} />
                <Text style={styles.name}>{user.email}</Text>
              </View>
            </View>
            <Separator style={{marginVertical: 20}} />
          </View>}

        {user.company &&
          user.company.address &&
          <View>
            <View style={styles.rowContainer}>
              <Text style={styles.label}>COMPANY ADDRESS</Text>
              <View style={styles.content}>
                <Ionicons
                  name="ios-pin-outline"
                  size={22}
                  style={styles.icon}
                />
                <Text style={styles.name}>{user.company.address}</Text>
              </View>
            </View>
            <Separator style={{marginVertical: 20}} />
          </View>}

        {user.company &&
          user.company.description &&
          <View>
            <View style={styles.rowContainer}>
              <Text style={styles.label}>COMPANY DESCRIPTION</Text>
              <View style={styles.content}>
                <Ionicons name="md-bulb" size={22} style={styles.icon} />
                <Text style={styles.name}>{user.company.description}</Text>
              </View>
            </View>
            <Separator style={{marginVertical: 20}} />
          </View>}

      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    backgroundColor: 'white',
    paddingTop: 20,
  },
  description: {
    fontWeight: '100',
    color: colors.darkGrey,
    fontSize: 15,
  },
  content: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  name: {
    fontSize: 16,
    fontWeight: '100',
    color: colors.darkGrey,
    paddingHorizontal: 5,
  },
  label: {
    paddingBottom: 5,
    color: colors.smokeGreyLight,
  },
  icon: {
    width: 20,
    height: 20,
  },
});
