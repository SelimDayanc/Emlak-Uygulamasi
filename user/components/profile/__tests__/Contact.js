import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Contact from './../Contact';

test('renders heart', () => {
  const tree = renderer
    .create(<Contact user={{name: 'name', email: 'z4ls@live.com'}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
