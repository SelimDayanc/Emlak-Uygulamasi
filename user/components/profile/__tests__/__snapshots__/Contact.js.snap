// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<RCTScrollView
  alwaysBounceHorizontal={undefined}
  alwaysBounceVertical={true}
  onContentSizeChange={null}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={undefined}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  sendMomentumEvents={false}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "padding": 10,
        "paddingTop": 20,
      },
    ]
  }
>
  <RCTScrollContentView
    collapsable={false}
    removeClippedSubviews={undefined}
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e3e7ed",
              "paddingBottom": 5,
            }
          }
        >
          EMAIL
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 22,
                },
                Object {
                  "height": 20,
                  "width": 20,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ï†¸
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#484848",
                "fontSize": 16,
                "fontWeight": "100",
                "paddingHorizontal": 5,
              }
            }
          >
            z4ls@live.com
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e3e7ed",
              "height": 0.5,
            },
            Object {
              "marginVertical": 20,
            },
          ]
        }
      />
    </View>
  </RCTScrollContentView>
</RCTScrollView>
`;
